### ISSUE TITLE
Update OEP-18 Python Dependency Management with more constraint tips

### ISSUE THREAD

**Issue by robrap** [openedx/open-edx-proposals#332]
In order to help with unpinning pinned constraints, we would like to propose the following updates when pinning constraints:
1. Include the pinning date in the comment for the pin. This will make it simpler to know how out-of-date we are.
2. Create a github issue for unpinning the constraint.

**Comment by robrap** [comment]
I believe this would be helpful, and should be relatively simple. I captured it as an issue, rather than opening a PR, because I just didn't have the time to deal with the edits and see it through.

**Comment by sarina** [comment]
Makes sense, thanks. I'd like to see what @feanil and/or other members of Maintainers WG feels about this - I don't think I'm the best person to propose this change, as I am not deeply maintaining code repos, but I can help this process move forward if the Maintainers WG wants to push it forward.

**Comment by kdmccormick** [comment]
Big +1 to the GitHub issue idea. Constraints are always a problem during upgrades, so having documentation and accountability for any constraint seems like a solid best practice.

My instinct would be to put the date in the GitHub issue rather than the comment, but that's not a strongly-held opinion. I'd support the change either way.

**Comment by robrap** [comment]
> My instinct would be to put the date in the GitHub issue rather than the comment, but that's not a strongly-held opinion. I'd support the change either way.

My thinking on the dates was simply that when looking over the constraints files, seeing that a constraint was added 5 years ago might stand-out and might get some action. Git blame is more for when you know what you are looking for, and dates on issues requires click-through. Anyway - just a thought.

> I think I could take or leave aspirational dates

There should be nothing aspirational about the date. It is simply the date on which the constraint was pinned/added.

**Comment by sarina** [comment]
With three votes, but from people who've thought about this a lot, I think this is a good recommendation to move forward with.

@feanil I agree with Robert there's nothing in the issue description about an aspirational date of removal. This is just about documenting _when_ the constraint was added, by both a comment and an associated GitHub issue.

I'm marking this as the Help Wanted label for anyone in the community to pick up, but I will try to draw down open issues as well over the next months.

**Comment by robrap** [comment]
Note that equally important to documenting this would be putting it into practice in edx-platform and the common constraints file. We could chip away at this, but it might help to make issues for these two repos.

**Comment by sarina** [comment]
@robrap I see https://github.com/openedx/edx-platform/issues/35055 was addressed and I see dates in https://github.com/openedx/edx-platform/blob/master/requirements/constraints.txt - yay!

Would the best way for me to go about addressing this issue to update the OEP with this recommendation? In [OEP-18](https://docs.openedx.org/projects/openedx-proposals/en/latest/best-practices/oep-0018-bp-python-dependencies.html) I'm not sure where the best place to add this in, but possibly step 2 in Specification and/or within the "Imposing Constraints on Dependencies" section?

**Comment by robrap** [comment]
@sarina: I'm all for you updating the OEP. I'd include a little code example with whatever text. Thank you.
